docs/reference/administration/system-options: add interface option (#277)
canonical/snap-docs: docs/reference/administration/system-options.md

> ⓘ Available since snapd _2.59_.

## system interface

> Allows configuring certain behaviour of individual interface rules. For now, the only allowed option is to configure the auto-connection.

Reviewer: ```suggestion
Permits specific interfaces options to be changed . The only option currently supported is interface [auto-connection](https://snapcraft.io/docs/explanation/interfaces/interface-auto-connection/).
```

> - **.allow-auto-connection**: configure auto-connection rules for the interface, options are `{false|true|verified}`.

Reviewer: ```suggestion
- **interface.allow-auto-connection**: configure auto-connection rules for the interface, options are `{false|true|verified}`.
```

(we've used this pattern elsewhere as the dot is easily missed)

>   * `false` or `true` controls whether any auto-connection is allowed at all.

Reviewer: ```suggestion
  * `false` or `true` controls whether auto-connections are permitted.
```

  * `verified` means that only snaps carrying a snap declaration can auto-connect this interface.

> The following example would configure the `x11` interface

Reviewer: _disallow_ is a little unusual - would something like block, reject, or deny be just as accurate?

Author: Good point, changed to 'deny' as that feels more in line
